File: android/src/main/java/io/mosip/tuvali/wallet/characteristics/VerificationStatusCharacteristic.kt
package io.mosip.tuvali.wallet.characteristics

import io.mosip.tuvali.transfer.CRCValidator
import io.mosip.tuvali.verifier.GattService
import java.util.*

class VerificationStatusCharacteristic(val data: ByteArray) {
  val status = data[0].toInt()


  companion object{
    val uuid: UUID = GattService.VERIFICATION_STATUS_CHAR_UUID
  }

  init{
    CRCValidator.validateCrcSentByVerifier(data, javaClass.simpleName)
  }
}

File: ios/Wallet/characteristics/VerificationStatusCharacteristic.swift

import Foundation
import CoreBluetooth


class VerificationStatusCharacteristic {
  var data: Data
  static let uuid = CharacteristicIds.VERIFICATION_STATUS_CHAR_UUID
  var status : Int

  init(data: Data){
      self.data =  data
      status = Int(data[0])
      CRCValidator.verify(data: data, characteristic: "VerificationStatusCharacteristic")
  }
}
